### PR TITLE
Add scrollspy on sidebars and TOCs

### PR DESCRIPTION
**Pull request by yogeshjain999**
1. Show right TOCs as per active section
2. Add scrollspy on left navigation to show active section
3. Add scrollspy on right TOC to show active sub-section

https://github.com/apotonick/trailblazer-front/assets/6301525/f1a477e3-b2d7-4e33-8e50-2fe7fde4592d

### CODE REVIEW

File: app/concepts/cell/documentation/documentation.erb [hunk 1]
```diff
@@ -1,46 +1,39 @@
...
         </div>
-        <%= tocs_right %>
+        <div class="relative" id="right-toc">
+          <%= tocs_right %>
+        </div>
       </div>
     </div>
   </div>
 </section>
-<section class="px-10 py-22 bg-bg-purple-3">
```
Comment (yogeshjain999) [inline app/concepts/cell/documentation/documentation.erb:36]: @apotonick I removed the call out section at the bottom as it was causing some UI conflicts. IMO, it seems redundant given we've same button in top nav on the screen. Do you think we should keep it ?

Screenshot 2023-09-02 at 10 12 27 PM

File: app/concepts/cell/documentation/documentation.erb [hunk 2]
```diff
@@ -1,46 +1,39 @@
...
   <div class="lg:flex">
     <nav class="documentation-side-nav" id="sideNav">
-      <div class="lg:hidden absolute right-4 top-4 text-3xl text-white" id="sideNavHideButton">
+      <div class="absolute text-3xl text-white lg:hidden right-4 top-4" id="sideNavHideButton">
         X
       </div>
       <%= toc_left %>
     </nav>
-    <div class="lg:p-10 pl-8 pr-4 py-10 bg-light-grey grow">
-      <h1 class="lg:text-3xl text-2xl text-blue font-bold">
```
Comment (yogeshjain999) [inline app/concepts/cell/documentation/documentation.erb:13]: @apotonick I also had to remove the gray area around top-level header as it was not getting aligned with right TOC when page gets scrolled. Instead, kept all background to white and added border around main content if that's fine.

https://github.com/apotonick/trailblazer-front/assets/6301525/36bfd337-3fff-4da3-9ab9-616e5fc73c2a

File: app/concepts/cell/documentation/documentation.erb (hunk 1 above)
Comment (apotonick) [reply app/concepts/cell/documentation/documentation.erb:36]: We should keep that and rather fix the menu to not slide into it?

File: app/concepts/cell/documentation/documentation.erb (hunk 2 above)
Comment (apotonick) [reply app/concepts/cell/documentation/documentation.erb:13]: Please revert to the designer's original version, the page looks very different without the gray parts.

File: app/concepts/cell/documentation/documentation.erb (hunk 1 above)
Comment (yogeshjain999) [reply app/concepts/cell/documentation/documentation.erb:36]: Sure, I've added this section back.

**Comment by apotonick** [comment]
Amazing job! :green_heart:

**Pull request merged**